### ISSUE TITLE
add a function to the model and trainer class to print the current hyperparameters of each model, trainer

### ISSUE THREAD

**Issue by smilesun** [marrlab/DomainLab#759]

**Comment by MatteoWohlrapp** [comment]
Implemented the functionality; currently, printouts include all information about the object and look like the print below. We could apply some filtering for keywords before printing to limit verbosity, just let me know the key metrics you want to see.

Parameters of TrainerDIAL: {'_task': None, '_success_node': <domainlab.algos.trainers.train_basic.TrainerBasic object at 0x7fbcbb2c1880>, '_parent_node': <domainlab.algos.trainers.train_matchdg.TrainerMatchDG object at 0x7fbcbb2c1a00>, 'next_model': None, '_model': ModelDAN(
  (_net_classifier): ClassifDropoutReluLinear(
    (op_drop): Dropout(p=0.5, inplace=False)
    (op_linear): Linear(in_features=2, out_features=2, bias=True)
  )
  (loss4gen_adv): KLDivLoss()
  (_net_invar_feat): NetConvBnReluPool2L(
    (conv_net): Sequential(
      (0): Conv2d(3, 32, kernel_size=(5, 5), stride=(1, 1), bias=False)
      (1): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (2): ReLU()
      (3): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
      (4): Conv2d(32, 64, kernel_size=(5, 5), stride=(1, 1), bias=False)
      (5): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (6): ReLU()
      (7): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
    )
    (layer_last): Linear(in_features=179776, out_features=2, bias=True)
  )
  (net_discriminator): ClassifDropoutReluLinear(
    (op_drop): Dropout(p=0.5, inplace=False)
    (op_linear): Linear(in_features=2, out_features=2, bias=True)
  )
), '_decoratee': None, 'task': <domainlab.tasks.task_folder_mk.mk_task_folder.<locals>.NodeTaskFolderDummy object at 0x7fbcbb2cb910>, 'observer': <domainlab.algos.observers.c_obvisitor_cleanup.ObVisitorCleanUp object at 0x7fbcbb2cb580>, 'device': device(type='cpu'), 'aconf': Namespace(lr=0.0001, gamma_reg={'default': 5.0, 'dann': 1.0, 'diva': 2.0}, es=1, seed=0, nocu=False, device=None, gen=False, keep_model=False, epos=2, epos_min=0, epo_te=1, warmup=100, debug=False, dmem=False, no_dump=False, trainer='dial', out='zoutput', dpath='zdpath', tpath=None, npath=None, npath_dom=None, npath_argna2val=None, nname_argna2val=None, nname='conv_bn_pool_2', nname_dom=None, apath=None, exptag='exptag', aggtag='aggtag', bm_dir=None, plot_data=None, outp_dir='zoutput/benchmarks/shell_benchmark', param_idx=True, msel='val', model='dann', acon=None, task='mini_vlcs', val_threshold=None, bs=100, split=0, te_d=['caltech'], tr_d=None, san_check=False, san_num=8, loglevel='DEBUG', shuffling_off=False, zd_dim=64, zx_dim=0, zy_dim=64, topic_dim=3, nname_encoder_x2topic_h=None, npath_encoder_x2topic_h=None, nname_encoder_sandwich_x2h4zd=None, npath_encoder_sandwich_x2h4zd=None, gamma_y=700000.0, gamma_d=100000.0, beta_t=1.0, beta_d=1.0, beta_x=1.0, beta_y=1.0, tau=0.05, epos_per_match_update=5, epochs_ctr=1, nperm=31, pperm=0.1, jigen_ppath=None, grid_len=3, dial_steps_perturb=3, dial_noise_scale=0.001, dial_lr=0.003, dial_epsilon=0.031), 'dict_loader_tr': {'sun': <torch.utils.data.dataloader.DataLoader object at 0x7fbcd9a8c460>, 'labelme': <torch.utils.data.dataloader.DataLoader object at 0x7fbcbb2d3700>}, 'loader_tr': <torch.utils.data.dataloader.DataLoader object at 0x7fbcbb2d3940>, 'loader_te': <torch.utils.data.dataloader.DataLoader object at 0x7fbcbb2d3d00>, 'num_batches': 1, 'flag_update_hyper_per_epoch': False, 'flag_update_hyper_per_batch': False, 'epo_loss_tr': 0, 'epo_reg_loss_tr': [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 'epo_task_loss_tr': 0, 'counter_batch': 0.0, 'hyper_scheduler': None, 'optimizer': Adam (
Parameter Group 0
    amsgrad: False
    betas: (0.9, 0.999)
    capturable: False
    eps: 1e-08
    foreach: None
    lr: 0.0001
    maximize: False
    weight_decay: 0
), 'exp': None, 'lambda_ctr': None, 'flag_stop': None, 'flag_erm': None, 'tensor_ref_domain2each_domain_x': None, 'tensor_ref_domain2each_domain_y': None, 'base_domain_size': None, 'tuple_tensor_ref_domain2each_y': None, 'tuple_tensor_refdomain2each': None, 'inner_trainer': None, 'loader_tr_source_target': None, 'flag_initialized': True}
